test/node/bigint_tests.ts

```diff
@@ -0,0 +1,123 @@
+import * as BSON from '../register-bson';
+import * as byteUtils from './tools/utils';
+import * as CONSTANTS from '../../src/constants';
+
+describe('BSON BigInt serialization Support', function () {
+  // Index for the data type byte of a BSON document with a single element
+  const DATA_TYPE_OFFSET = 4;
+  before(function () {
+    try {
+      BigInt(0);
+    } catch (_) {
+      this.skip('JS VM does not support BigInt');
+    }
+  });
+
+  it('Serializes bigints with the correct BSON type', function () {
+    const testDoc = { a: 0n };
+    const serializedDoc = BSON.serialize(testDoc);
+    expect(serializedDoc[DATA_TYPE_OFFSET]).to.equal(CONSTANTS.BSON_DATA_LONG);
+  });
+
+  it('Serializes bigints into little-endian byte order', function () {
+    const testDoc = { a: BigInt(0x1234567812345678n) };
```

Comment:
```suggestion
    const testDoc = { a:0x1234567812345678n };
```

No need for the cast and `n` syntax, just redundant, I won't comment on every instance but there are more